src/backend/codegen/codegen_manager.cc

```diff
@@ -0,0 +1,90 @@
+//---------------------------------------------------------------------------
+//  Greenplum Database
+//  Copyright (C) 2016 Pivotal Software, Inc.
+//
+//  @filename:
+//    codegen_manager.cpp
+//
+//  @doc:
+//    Implementation of a code generator manager
+//
+//---------------------------------------------------------------------------
+
+#include <cstdint>
+#include <string>
+
+#include "codegen/utils/clang_compiler.h"
+#include "codegen/utils/utility.h"
+#include "codegen/utils/instance_method_wrappers.h"
+#include "codegen/utils/codegen_utils.h"
+#include "codegen/codegen_interface.h"
+
+#include "codegen/codegen_manager.h"
+#include "llvm/ADT/APFloat.h"
+#include "llvm/ADT/APInt.h"
+#include "llvm/IR/Argument.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Constant.h"
+#include "llvm/IR/DerivedTypes.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/GlobalVariable.h"
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Type.h"
+#include "llvm/IR/Value.h"
+#include "llvm/IR/Verifier.h"
+#include "llvm/Support/Casting.h"
+
+using gpcodegen::CodeGenManager;
+
+CodeGenManager::CodeGenManager() {
+  codegen_utils_.reset(new gpcodegen::CodeGenUtils("test_module"));
```

Comment:
Let me change it to string as an argument in constructor and let operator to pass its name for that.